code/modules/mob/living/carbon/xenomorph/xenoprocs.dm

```diff
@@ -605,7 +605,7 @@
 		return FALSE
 	var/datum/reagent/body_tox
 	var/i = 1
-	do
+	while(i++ < count && do_after(src, channel_time, TRUE, C, BUSY_ICON_HOSTILE))
```

Comment:
does this not also affect defilers

Reply:
nop, this proc is used by drone and sentinel only